src/composablesTs/useMutationObserver.ts

```diff
@@ -1,41 +1,44 @@
-import { onBeforeUnmount, onMounted, toValue, watch } from "vue";
+import { computed, onBeforeUnmount, onMounted, toValue, watch } from "vue";
 import { isSSR } from "../utilsTs/utilHelper.ts";
 
-import type { Ref } from "vue";
+import type { TemplateRefElement } from "src/types.ts";
 
 export function useMutationObserver(
-  target: Ref<HTMLElement | null>,
-  callBack: () => void,
+  target: TemplateRefElement,
+  callback: MutationCallback,
   config = { childList: true, attributes: true, characterData: true },
 ) {
   if (isSSR) return;
 
-  const observer = new MutationObserver(callBack);
+  const observer = new MutationObserver(callback);
+  const targetValue = computed(() => toValue(target));
```

Comment:
what the reason to get value in a computed?

`toValue(target)` is same that `targetValue.value` but shorter.

Reply:
```
    if (Array.isArray(toValue(target))) {
      toValue(target).value.forEach((element) => {
        observer.observe(element, config);
      });
    } else {
      observer.observe(toValue(target), config);
    }
```
This code is not valid. toValue is function and it return a result each time, so values in If condition and  If block are not the same and typescript cannot be sure if it's array or not.

Reply:
Okay